### PR TITLE
add devise

### PR DESCRIPTION
**Pull request by leloufadel**
Description:
This pull request addresses the project requirements by implementing Devise for user authentication in the blog app. The changes include:

Removal of the current_user method from the ApplicationController as Devise will provide this functionality.
Installation and setup of Devise to handle user registration, login, email confirmation, and password reset.
Modification of Devise views for registration.

Capture-Do-solo-Project

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (Owusu-Desmond) [inline app/models/user.rb]: - [ ] Kindly ensure that you enabled `:confirmable` in the user model which should allow a sure to confirm the registration process below proceeding. You should see something similar to what is below: :point_down: 
  
  ```ruby
	  devise :database_authenticatable, :registerable,
				  :recoverable, :rememberable, :validatable, :confirmable
  ```

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  </div>
+
+  <% if devise_mapping.rememberable? %>
+    <div class="field">
+      <%= f.check_box :remember_me %>
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
```
Comment (Owusu-Desmond) [inline app/views/devise/sessions/new.html.erb]: - [ ] From the project requirement, you should modify the views in device for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A). So make sure that your login page match the wireframe. :pray: 

### Your login page
  image


### Wireframe

  image

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @leloufadel , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by joel-kalema — APPROVED** [review]
Hi @leloufadel 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 


## Highlights

The project is running successfully 👍 
PR is well descriptive 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@joel-kalema** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**